contracts/AcceleratingDistributor.sol

```diff
@@ -8,27 +8,47 @@ import "@openzeppelin/contracts/access/Ownable.sol";
- * only compatible with standard ERC20 tokens, and not tokens that charge fees on transfers, dynamically change
- * balance, or have double entry-points. It's up to the contract owner to ensure they only add supported tokens.
+ * @title Across Accelerating Distributor (Staking) Contract
+ * @notice Stakers start by earning their pro-rata share of a baseEmissionRate per second. The baseEmissionRate is
+ * amplified by a reward multiplier, which increases up to a configurable maxMultiplier, based on the time and amounts
+ * of previous deposits into the contract. Multiple LP tokens can be staked in this contract enabling depositors to
+ * batch stake and claim via multicall.
+ * @notice This contract is only compatible with standard ERC20 tokens, and not tokens that charge fees on transfers,
+ * dynamically change balance, or have double entry-points. It's the responsibility of the contract owner to ensure
+ * they only add supported tokens, and that staking token configurations are applied correctly.
+ * @dev This contract is inspired by the Synthetix staking contract, and the Ampleforth geyser.
  */
 
 contract AcceleratingDistributor is ReentrancyGuard, Ownable, Multicall {
     using SafeERC20 for IERC20;
 
     IERC20 public immutable rewardToken;
 
-    // Each User deposit is tracked with the information below.
+    /**
+     * @notice Tracks the staking balance and associated rewards of an address for a single staking token.
+     * @dev Stored user rewards are updated each time a Staker Function is called with the user's address.
+     * @param cumulativeBalance User's current total staking balance in the contract for this address.
+     * @param averageDepositTime Averaged timestamp of user's entry into the pool, weighted by the size of each deposit.
+     * @param rewardsAccumulatedPerToken User's cumulative per-unit share of staking rewards since last update.
+     * @param rewardsOutstanding Staking reward tokens available to be claimed since last update.
+     */
     struct UserDeposit {
         uint256 cumulativeBalance;
         uint256 averageDepositTime;
         uint256 rewardsAccumulatedPerToken;
         uint256 rewardsOutstanding;
     }
 
+    /**
+     * @notice Tracks the global configuration and state of each staking token.
+     * @dev Global stakingToken state is updated each time a Staker Function is called.
+     * @param baseEmissionRate Base staking token emission rate, before applying any user multiplier.
+     * @param maxMultiplier Maximum achievable multiplier to be applied to baseEmissionRate.
+     * @param secondsToMaxMulitplier Number of seconds after user's averageDepositTime before reaching maxMultiplier.
+     * @param cumulativeStaked Total amount of deposit token staked in contract.
+     * @param rewardPerTokenStored Global cumulative per-unit share of staking rewards since last update.
```

Comment:
Is `rewardPerTokenStored` cumulative since last update or cumulative period?

Reply:
Ah, good catch. What I am intending to say is that the value is cumulative since the token was configured, but that it was computed as at the last update. I think the word "since" is not correct in this context; could probably change it as follows: "since last update" => "as at last update". wdyt?

Reply:
yeah I think thats more accurate